nukebuild/BuildParameters.cs

```diff
@@ -0,0 +1,186 @@
+        public AbsolutePath NugetRoot { get; }
+        public AbsolutePath ZipRoot { get; }
+        public AbsolutePath BinRoot { get; }
+        public AbsolutePath TestResultsRoot { get; }
+        public string DirSuffix { get; }
+        public List<string> BuildDirs { get; }
+        public string FileZipSuffix { get; }
+        public AbsolutePath ZipCoreArtifacts { get; }
+        public AbsolutePath ZipNuGetArtifacts { get; }
+        public AbsolutePath ZipSourceControlCatalogDesktopDir { get; }
+        public AbsolutePath ZipTargetControlCatalogDesktopDir { get; }
+
+
+       public BuildParameters(Build b)
+        {
+            var buildSystem = Host;
+            
+
+            // ARGUMENTS
+            Configuration = b.NukeArgConfiguration ?? "Release";
+            SkipTests = b.NukeArgSkipTests;
+
+            // CONFIGURATION
+            MainRepo = "https://github.com/AvaloniaUI/Avalonia";
+            MasterBranch = "refs/heads/master";
+            ReleaseBranchPrefix = "refs/heads/release/";
+            ReleaseConfiguration = "Release";
+            MSBuildSolution = RootDirectory / "dirs.proj";
+
+            // PARAMETERS
+            IsLocalBuild = buildSystem == HostType.Console;
+            IsRunningOnUnix = Environment.OSVersion.Platform == PlatformID.Unix ||
+                              Environment.OSVersion.Platform == PlatformID.MacOSX;
+            IsRunningOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+            IsRunningOnAppVeyor = buildSystem == HostType.AppVeyor;
+            IsRunningOnAzure = buildSystem == HostType.TeamServices ||
+                               Environment.GetEnvironmentVariable("LOGNAME") == "vsts";
+
+            string tagName = null;
+            if (IsRunningOnAppVeyor)
+            {
+                IsPullRequest = AppVeyor.Instance.PullRequestNumber != 0;
+                RepositoryName = Environment.GetEnvironmentVariable("BUILD_REPOSITORY_URI");
+                RepositoryBranch = Environment.GetEnvironmentVariable("BUILD_SOURCEBRANCH");
+                
+                IsReleaseBranch =
+                    (Environment.GetEnvironmentVariable("BUILD_SOURCEBRANCH") ?? "").StartsWith(ReleaseBranchPrefix,
+                        StringComparison.OrdinalIgnoreCase);
+                IsTagged = AppVeyor.Instance.RepositoryTag
+                           && !string.IsNullOrWhiteSpace(AppVeyor.Instance.RepositoryTagName);
+                
+                tagName = AppVeyor.Instance.RepositoryTagName;
+            }
+            else if (IsRunningOnAzure)
+            {
+                RepositoryName = TeamServices.Instance.RepositoryUri;
+                RepositoryBranch = TeamServices.Instance.SourceBranch;
+                IsPullRequest = TeamServices.Instance.PullRequestId.HasValue;
+                IsMainRepo = StringComparer.OrdinalIgnoreCase.Equals(MainRepo, TeamServices.Instance.RepositoryUri);
+                
+                // TODO???
```

Comment:
What's this TODO for?

Reply:
I have found a way to get the tag from Azure DevOps. The original script was ignoring it though.